sample/src/SampleAppRedirectOnLaunch.js

```diff
@@ -109,29 +122,11 @@ class SampleAppRedirectOnLaunch extends React.Component {
           <div />
         )}
         <AzureAD
-          provider={
-            new MsalAuthProviderFactory(
-              {
-                auth: {
-                  authority: process.env.REACT_APP_AUTHORITY,
-                  clientId: process.env.REACT_APP_AAD_APP_CLIENT_ID,
-                  redirectUri: window.location.origin,
-                  postLogoutRedirectUri: window.location.origin,
-                },
-                cache: {
-                  cacheLocation: 'sessionStorage',
-                  storeAuthStateInCookie: true,
-                },
-              },
-              {
-                scopes: ['openid'],
-              },
-              LoginType.Redirect,
-            )
-          }
+          provider={authProviderFactory}
```

Comment:
After a lot of testing, I found issues with creating the provide inline like it had been.

1) This causes the AzureAD component to always re-render
2) It's possible to have multiple instances of the provider, which can cause issues in MSAL

I ended up putting it into a separate file to avoid errors in MSAL. Ultimately, this deserves some documentation in a troubleshooting section.